packages/clerk-js/src/core/fraudProtection.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * TODO: @nikos Move captcha and fraud detection logic to this class
+ */
+class FraudProtectionService {
+  private inflightRequest: Promise<unknown> | null = null;
+
+  public async execute<T extends () => Promise<any>>(cb: T): Promise<Awaited<ReturnType<T>>> {
+    if (this.inflightRequest) {
+      console.error('Clerk: This state should not be possible. Please report this to Clerk support.');
```

Comment:
Why is that ? Doesn't it make sense to queue callback here ? 

Will this cause 2 captcha modals back to back ?

Reply:
This is mostly for debugging, as this state should never be possible based on the changes in Session.ts
 We always  block on  `await fraudProtection.blockUntilReady();` and only after do we call `execute`